forem/forem
# Conditionally disabling Listings (Waiting on Product but technically Good to Go)

## What type of PR is this? (check all applicable)


## Description

Prior to this commit, we were assuming everyone still had the Listing's
feature turned on.

With this commit, we move from the "implicitly on" nature of
`FeatureFlag.accessible?` to the "explicitly off" nature of
`FeatureFlag.enabled?`.  The key differentiator is that we are not
"pushing out a feature flag setting to other forems."  Listings now
become an "opt-in".

This requires a few things:

1) Ensure that DEV.to enables the `:listing_feature` before we deploy
   this branch.
2) Communicate this change to other Forem adopters and provide a
   timeline for when we'll deploy this change.
3) Documentation.  Given that we have this feature, but it will be
   disabled by default, we want to look into communicating what it is
   and how to enable it.

Further, I chose to do the least effort change on our specs.  Namely to
treat the specs as though someone fully enabled the listing feature.


## Related Tickets & Documents

This pull request is a "competetor" to #16423.

Closes forem/rfcs#291

Relates to #16420 and #16416.

## QA Instructions, Screenshots, Recordings

The instructions of #16406 don't quite apply, but are similar:

Testing this change:

1.   Start your local application (e.g. $ bin/startup)
2.   Go to http://localhost:3000/listings
3.   You should get an ActiveRecord::RecordNotFound error; because the
     application is now looking for a user or org named "listings"; in other
     words we're not routing GET "/listings" to listings#index controller
     action.
4.   Now enable the feature (e.g. $ bin/rails r \ "FeatureFlag.enable(:listing_feature)")
5.   Refresh http://localhost:3000/listings
6.   You should get a Successful response.
8.   Now disable the feature (e.g. $ bin/rails r \ "FeatureFlag.disable(:listing_feature)")
8.   Refresh http://localhost:3000/listings
9.   You should get an ActiveRecord::RecordNotFound error
10.  Goto 4

### UI accessibility concerns?

## Added/updated tests?


## [Forem core team only] How will this change be communicated?
